cmd/k8s-operator/operator.go

```diff
@@ -317,12 +319,134 @@ func runReconcilers(zlog *zap.SugaredLogger, s *tsnet.Server, tsNamespace string
 	if err != nil {
 		startlog.Fatalf("could not create nameserver reconciler: %v", err)
 	}
+	lc, err := s.LocalClient()
+	if err != nil {
+		startlog.Fatalf("error retrieving local client: %w", err)
+	}
+	// On DNSConfig changes, reconcile all EndpointSlices in operator namespace.
+	dnsConfigFilter := handler.EnqueueRequestsFromMapFunc(enqueueAllEndpointSlicesInNS(tsNamespace, mgr.GetClient()))
+	// On Secret changes, if it has the tailscale labels and is for an
+	// ingress/egress proxy, reconcile the EndpointSlice for the proxy's
+	// headless Service. We need to watch Secrets because this is where the
+	// dns-records-reconciler reads the MagicDNS name from for ingress
+	// proxies exposed via an annotation.
+	epsForSecretFilter := handler.EnqueueRequestsFromMapFunc(enqueueEndpointSliceForSecret(tsNamespace, mgr.GetClient()))
+	// The only Service changes the dns-records-reconciler is interested in
+	// are changes to svc.status.loadBalancer.ingress.hostname, so only
+	// reconcile proxy EndpointSlices associated with LoadBalancer Services
+	// exposed via Tailscale.
+	epsForServiceFilter := handler.EnqueueRequestsFromMapFunc(enqueueEndpointSliceForService(tsNamespace, mgr.GetClient(), isDefaultLoadBalancer))
+	// If a tailscale Ingress changes, reconcile the EndpointSlice for the proxy's headless Service.
+	epsForIngressFilter := handler.EnqueueRequestsFromMapFunc(enqueueEndpointSliceForIngress(tsNamespace, mgr.GetClient()))
+	err = builder.ControllerManagedBy(mgr).
+		For(&discoveryv1.EndpointSlice{}).
```

Comment:
We reconcile EndpointSlice that contains the proxy Pod IP address(es). This is the one resource that allows us to trigger an update for one particular proxy